ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java

```diff
@@ -14,98 +14,39 @@
  */
 package org.hyperledger.besu.ethereum.api.jsonrpc.internal;
 
-import org.hyperledger.besu.ethereum.api.jsonrpc.internal.exception.InvalidJsonRpcRequestException;
-import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.JsonRpcParameter;
-
-import java.util.Arrays;
 import java.util.Objects;
 import java.util.Optional;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonGetter;
-import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.annotation.JsonInclude.Include;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import com.fasterxml.jackson.annotation.JsonSetter;
+import io.vertx.ext.auth.User;
 
-@JsonIgnoreProperties(ignoreUnknown = true)
 public class JsonRpcRequest {
 
-  private final JsonRpcParameter parameterAccessor = new JsonRpcParameter();
-
-  private JsonRpcRequestId id;
-  private final String method;
-  private final Object[] params;
-  private final String version;
-  private boolean isNotification = true;
-
-  @JsonCreator
-  public JsonRpcRequest(
-      @JsonProperty("jsonrpc") final String version,
-      @JsonProperty("method") final String method,
-      @JsonProperty("params") final Object[] params) {
-    this.version = version;
-    this.method = method;
-    this.params = params;
-    if (method == null) {
-      throw new InvalidJsonRpcRequestException("Field 'method' is required");
-    }
-  }
-
-  @JsonGetter("id")
-  public Object getId() {
-    return id == null ? null : id.getValue();
-  }
-
-  @JsonGetter("method")
-  public String getMethod() {
-    return method;
-  }
+  private final JsonRpcRequestBody requestBody;
+  private final Optional<User> user;
 
-  @JsonGetter("jsonrpc")
-  public String getVersion() {
-    return version;
+  public JsonRpcRequest(final JsonRpcRequestBody requestBody) {
+    this(requestBody, Optional.empty());
   }
 
-  @JsonInclude(Include.NON_NULL)
-  @JsonGetter("params")
-  public Object[] getParams() {
-    return params;
+  public JsonRpcRequest(final JsonRpcRequestBody requestBody, final Optional<User> user) {
+    this.requestBody = requestBody;
+    this.user = user;
   }
 
-  @JsonIgnore
-  public boolean isNotification() {
-    return isNotification;
+  public JsonRpcRequestBody getRequestBody() {
```

Comment:
I'd suggest keeping the `requestBody` encapsulated and proving the request through to the `requestBody` (i.e mediator pattern)

```
public Object getId(){
  return requestBody.getId();
}
```

Reply:
move JsonRpcRequest to JsonRpcRequestContext, making this irrelevant (I think?) - ok?

Reply:
👍